tcl/auth.tcl

```diff
@@ -76,7 +76,7 @@ proc qc::auth_session { session_id } {
     #| Try to authenticate user based on the session_id given
     #| Return the user_id if successful
     #| On failure throw AUTH error
-    if { [session_exists $session_id] } {
+    if { [session_valid $session_id] } {
         return [session_user_id $session_id]
```

Comment:
If we make the change to ```qc::auth_session``` there will be ramifications for projects like tlc-www & tlc-erp but if we allow the developer to override ```qc::auth_session```'s default ```age_limit``` and ```idle_timeout``` I think that would be ok.

something like:

```
proc qc::session_valid {args} {
    #| Check session exists and has not expired.
    qc::args $args -age_limit ? -idle_timeout "1 hour" -- session_id

    if { [qc::param_exists session_age_limit] } {
        default age_limit [qc::param_get session_age_limit]
    } else {
        deafult age_limit "12 hours"
    }
    if { [qc::param_exists session_idle_timeout] } {
        default idle_timeout [qc::param_get session_idle_timeout]
    } else {
        deafult idle_timeout "1 hour"
    }

    set qry {
        select
        user_id
        from session
        where
        session_id=:session_id
        and (current_timestamp-time_modified) <=:idle_timeout::interval
        and (current_timestamp-time_created) <=:age_limit::interval
    }

    db_0or1row $qry {
        return false
    } {
        return true
    }
}
```